versionutils/repo.go

```diff
@@ -155,3 +157,19 @@ func ParseFullTomlFromDir(relativeDir string) (*TomlWrapper, error) {
 func ParseFullToml() (*TomlWrapper, error) {
 	return ParseFullTomlFromDir("")
 }
+
+func getVersionFromTree(tomlTree *toml.Tree, pkgName string) (found bool, version string) {
+	isEmpty := func(node *toml.Tree, key string) bool {
+		return node.Get(key) == nil || node.Get(key) == ""
+	}
+
+	switch {
+	case tomlTree.Get(nameConst) == pkgName && !isEmpty(tomlTree, versionConst):
+		return true, tomlTree.Get(versionConst).(string)
+	case tomlTree.Get(nameConst) == pkgName && !isEmpty(tomlTree, revisionConst):
+		return true, tomlTree.Get(revisionConst).(string)
+	case tomlTree.Get(nameConst) == pkgName && !isEmpty(tomlTree, branchConst):
```

Comment:
can we stop repeating `tomlTree.Get(nameConst) == pkgName` in the switch statement by checking this before the switch statement?